### DISCUSSION TITLE
Main image for Entitiy

### DISCUSSION THREAD

**Discussion by marcus-at-localhost** [atrocore/atrocore#673] [Q&A] [answered]
I've setup my own entity and created a 1:n relation to assets, but it seems like I can't set a main image.

I worked until a few weeks ago I think (to get an image in a API call that returns a list of an entity. Can you please give a guidance to do that? That would be great! Thanks!

I created a test case in the demo

https://demo.atropim.com/#Profile
https://demo.atropim.com/#Admin/fieldManager/scope=Profile
https://demo.atropim.com/#Admin/linkManager/scope=Profile

**Comment by atrolex** [comment, marked as answer]
Hello Marcus, we decided to deprecate this feature for several reasons. Now only products and categories have "main image", as a pre-coded additional business logic. 
As a workaround you can add a new field to the entity assets, to mark such asset. Alternatively, if the module advanced data types is used, a user can create such an entity relationship construction:
Some_entity {one-to-many} some_entity_asset {many- to-one} asset.
Some_entity_asset is an entity of type "relationship", so that you can add any field to it, of any data type, not only marking for the main image.
Such a construct is a technical implementation of being able to have properties for relations, in your case for the relation between some_entity and asset.

**Reply by marcus-at-localhost** [reply]
Thanks @atrolex for giving some insight. For now I found a workaround, but will explore what you offered as a solution.
I just wished I knew that in advanced.
I know the development is very active and I try to keep up with reading the changelog before every update, but since it's 3 repos and it's not always clear how changes in atrocore affect changes in atropim, it's always a bit of a gamble for me.
It was mentioned at some time that you plan to have a unified changelog somewhere. Does that already exist? That would be helpful.
Thanks for the great work!

**Reply by atrolex** [reply]
We usually don't remove features, to avoid situations like this, but this feature would be an appendix, which would disrupt the standard functionality. In short we have exchanged a lot of custom functionality of PIM with standard functionality of AtroCore. Some things are just done, if we see some problems. It hard to plan something like that in advance.
